libraries/AP_Terrain/TerrainIO.cpp

```diff
@@ -178,8 +178,11 @@ void AP_Terrain::open_file(void)
     // create directory if need be
     if (!directory_created) {
         *p = 0;
-        mkdir(file_path, 0755);
-        directory_created = true;
+        directory_created = !mkdir(file_path, 0755);
+        // if we didn't succeed at making the directory, then IO failed
+        if(!directory_created) {
+            io_failure = true;
```

Comment:
I think it doesn't matter as `p` is a local variable that will be gone after it returns.